tests/test_api_types.py

```diff
@@ -18,46 +18,145 @@
 
 def test_is_array_like() -> None:
     check(assert_type(api.is_array_like(arr), bool), bool)
+    check(assert_type(api.is_array_like(np.array([1, 2, 3])), bool), bool)
+    check(assert_type(api.is_array_like(np.dtype(np.int32)), bool), bool)
+    check(assert_type(api.is_array_like(pd.Series([1, 2, 3])), bool), bool)
+    check(
+        assert_type(api.is_array_like(pd.DataFrame({"a": [1, 2], "b": [3, 4]})), bool),
+        bool,
+    )
+    check(assert_type(api.is_array_like(pd.Index([1, 2.0])), bool), bool)
```

Comment:
can you change all of these to use a module-level variable.  Right now, we have at the top of this file:
```python
nparr = np.array([1, 2, 3])
arr = pd.Series([1, 2, 3])
obj = "True"
mapping = {"a": "a"}
```
so you can add something for `np.dtype(np.int32)` , `(pd.DataFrame({"a": [1, 2], "b": [3, 4]})` and `pd.Index([1, 2.0])` up top, and reuse those variables throughout.

Reply:
sure sir i'll do it

Reply:
Done the changes sir